src/ScriptEngine/ScriptEngine.cs

```diff
@@ -114,60 +121,88 @@ private void LoadDLL(string path, GameObject obj)
             if (!QuietMode.Value)
                 Logger.Log(LogLevel.Info, $"Loading plugins from {path}");
 
-            using (var dll = AssemblyDefinition.ReadAssembly(path, new ReaderParameters { AssemblyResolver = defaultResolver }))
+            using (var dll = AssemblyDefinition.ReadAssembly(path, new ReaderParameters {
+                AssemblyResolver = defaultResolver,
+                ReadSymbols = true
+            }))
             {
                 dll.Name.Name = $"{dll.Name.Name}-{DateTime.Now.Ticks}";
+                Assembly ass;
 
-                using (var ms = new MemoryStream())
+                if (DumpAssemblies.Value)
                 {
-                    dll.Write(ms);
-                    var ass = Assembly.Load(ms.ToArray());
+                    // Dump assembly & load it from disk
+                    if (!Directory.Exists(DumpedAssembliesPath))
+                        Directory.CreateDirectory(DumpedAssembliesPath);
+   
+                    string assemblyDumpPath = Path.Combine(DumpedAssembliesPath, dll.Name.Name + Path.GetExtension(dll.MainModule.Name));
 
-                    foreach (Type type in GetTypesSafe(ass))
+                    using (FileStream outFileStream = new FileStream(assemblyDumpPath, FileMode.Create))
                     {
-                        try
+                        dll.Write((Stream)outFileStream, new WriterParameters()
                         {
-                            if (!typeof(BaseUnityPlugin).IsAssignableFrom(type)) continue;
+                            WriteSymbols = true
+                        });
+                    }
+                    Logger.Log(LogLevel.Info, $"Dumped Assembly & Symbols to {assemblyDumpPath}");
 
-                            var metadata = MetadataHelper.GetMetadata(type);
-                            if (metadata == null) continue;
+                    ass = Assembly.LoadFile(assemblyDumpPath);
+                    Logger.Log(LogLevel.Info, $"Loaded dumped Assembly from {assemblyDumpPath}");
```

Comment:
Maybe collapse this into a single log entry with Debug level to not spam the log as much when there's multiple plugins reloaded.

Reply:
Yup that's a good idea :)